src/Compilers/CSharp/Portable/Symbols/SubstitutedMethodSymbol.cs

```diff
@@ -416,8 +416,7 @@ public override bool Equals(object obj)
             int arity = this.Arity;
             for (int i = 0; i < arity; i++)
             {
-                // TODO: what about annotations
-                if (!TypeSymbol.Equals(this.TypeArgumentsWithAnnotations[i].Type, other.TypeArgumentsWithAnnotations[i].Type, TypeCompareKind.ConsiderEverything2))
+                if (!this.TypeArgumentsWithAnnotations[i].Equals(other.TypeArgumentsWithAnnotations[i], TypeCompareKind.ConsiderEverything2))
```

Comment:
>ConsiderEverything2 [](start = 120, length = 19)

Can you remove the `2` if this is indeed the right comparison kind?
(consider doing the same a few lines above if you can also confirm, thanks)